packages/story-editor/src/components/canvas/utils/useSnapping.js

```diff
@@ -45,15 +45,17 @@ function useSnapping({
       pageHeight,
     })
   );
-  const { activeDropTargetId } = useDropTargets((state) => ({
-    activeDropTargetId: state.state.activeDropTargetId,
-  }));
+  const { activeDropTargetId, isDropTargetingDisabled } = useDropTargets(
+    (state) => ({
+      activeDropTargetId: state.state.activeDropTargetId,
+      isDropTargetingDisabled: state.state.isDropTargetingDisabled,
+    })
+  );
 
   const triggerOnboarding = useUserOnboarding(({ SAFE_ZONE }) => SAFE_ZONE);
 
-  // ⌘ key disables snapping
-  const snapDisabled = useGlobalIsKeyPressed('meta');
-  canSnap = canSnap && !snapDisabled && !activeDropTargetId;
+  // Drop-targeting is disabled with ⌘ key, we also disable snapping in this case.
+  canSnap = canSnap && !isDropTargetingDisabled && !activeDropTargetId;
```

Comment:
Using `useGlobalIsKeyPressed` doesn't work when used in 2 separate places, so opted for moving it to the drop-targeting provider (since that's accessible in both cases).